src/core/nexus/HY_PointHydroNexusRemote.cpp

```diff
@@ -40,7 +40,41 @@ HY_PointHydroNexusRemote::HY_PointHydroNexusRemote(int nexus_id_number, std::str
 
 HY_PointHydroNexusRemote::~HY_PointHydroNexusRemote()
 {
-    //dtor
+    long wait_time = 0;
+
+    // This destructore might be called after MPI_Finalize so do not attempt communication if
+    // this has occured
+    int mpi_finalized;
+    MPI_Finalized(&mpi_finalized);
+
+    while ( (stored_recieves.size() > 0 || stored_sends.size() > 0) && !mpi_finalized )
+    {
+        process_communications();
+
+        std::this_thread::sleep_for(std::chrono::milliseconds(500));
+
+        wait_time += 500;
+
+        if ( wait_time > 120000 )
```

Comment:
We should set this wait time as a macro at least, or better yet as a user configuration option.

Reply:
We could make both values class members and provide get and set functions